### DISCUSSION TITLE
Create a refreshAccessToken

### DISCUSSION THREAD

**Discussion by netdesignr** [supabase/supabase#10420] [Questions] [answered]
Hi,

Can someone give me a clue about how a refreshAccessToken function can be created using the new `v2`?

This is how I currently have it setup. I looked around the API/docs and can't seem to find the right indications for the upgrade.

```js
async function refreshAccessToken(refreshToken?: string) {
  if (!refreshToken) return null;

  const { data, error } = await supabaseAdmin.auth.api.refreshAccessToken(
    refreshToken
  );

  if (!data || error) return null;

  return mapAuthSession(data);
}
```

Thank you!

**Comment by GaryAustin1** [comment, marked as answer]
Maybe this:  https://supabase.com/docs/reference/javascript/auth-refreshsession 
You can pass in the refresh token.

**Reply by netdesignr** [reply]
Actually this just works as expected. I wasn't sure if this was the recommened way but actually this makes sense
